### ISSUE TITLE
Change host display refresh rate to follow content video mode

### ISSUE THREAD

**Issue by mgtroyas** [dosbox-staging/dosbox-staging#2312]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

windows-x86_64-v0.80.1

### What Operating System are you using?

Windows 11

### Is your feature request related to a problem? Please describe.

I have two monitors connected to muy computer, one is a modern LCD with VRR support (AMD Freesync), the other a 17 inch CRT.

I'm using Pinball Dreams as a test of smooth scrolling. It seems to use 70Hz refresh rate natively.

On my LCD, set to 75Hz refresh rate, the scroll is smooth because of VRR. But on my CRT, the scroll is only smooth if i set it to 70Hz. At 60Hz, 100Hz or 120Hz it's not smooth.

I use fullresolution=original so my CRT sets the resolution the game requests, but the refresh rate is not being changed, the one configured before launching DosBox is bein maintained on all resolutions.

### Describe the solution you'd like

I suggest honoring the refresh rate the game is trying to set, when changing video mode of the host.

### Describe alternatives you've considered

I suppose it's not being done to avoid problems with low refresh rate LCDs, but should be an option on higher refresh rate capable monitors, either auto detecting video modes availables or via a max refresh rate configuration setting.

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by Kappa971** [comment]
I'm not a developer of DOSBox Staging but I helped test it on VRR screens.
The current behavior of DOSBox Staging on a VRR screen (Freesync or G-Sync) is correct.
I have a 144hz Freesync screen and DOSBox shouldn't change the real screen refresh rate, it would break the Freesync. On the other hand the refresh rate of the DOSBox "virtual" screen changes correctly according to what the game requires, and the VRR monitor adapts to the emulator's fps.
Perhaps with `fullresolution=original`, DOSBox should be able to change the real screen refresh rate as well so that you can use it on a CRT screen, without affecting the option `fullresolution=desktop` for VRR LCD/OLED screens, but that's just the devs they will be able to tell if it is feasible.

**Comment by mgtroyas** [comment]
Yes, on my VRR LCD monitor it works that way, and the scrolling is butter smooth, the problem arises when using the classic CRT.

No, unfortunately using `fullresolution=original` doesn't change screen refresh rate when changing resolution, it mantains the previous one. That's why I'm opening this issue. 

Retroarch changes both resolution and refresh rate without problems on this same setup, so it's not a operating system/drivers limitation. Perhaps a SDL one, I don't know, that's why I'm asking.

**Comment by johnnovak** [comment]
> Yes, on my VRR LCD monitor it works that way, and the scrolling is butter smooth, the problem arises when using the classic CRT.
> 
> No, unfortunately using `fullresolution=original` doesn't change screen refresh rate when changing resolution, it mantains the previous one. That's why I'm opening this issue.
> 
> Retroarch changes both resolution and refresh rate without problems on this same setup, so it's not a operating system/drivers limitation. Perhaps a SDL one, I don't know, that's why I'm asking.

It seems reasonable for us to do the same thing then. It definitely sounds like a shortcoming of DOSBox.

**Comment by GranMinigun** [comment]
Yeah, it's fairly simple to do... Have to stop fighting SDL and just set a fullscreen display mode separately from window size.

**Comment by johnnovak** [comment]
Based on my recent learning in relation to WinUAE, I'm wondering if this is possible at all in Windows 10+ because of the full screen optimisation thing. See my problem here:
https://github.com/tonioni/WinUAE/issues/253

This is a verygood article about the subject:
https://devblogs.microsoft.com/directx/demystifying-full-screen-optimizations/

Right now I just set my custom created 70Hz resolution before starting a DOS game where the 60 vs 70 Hz difference matters (for me that's a very short list because Im not much into action games).

**Comment by kcgen** [comment]
Some 60 Hz displays let you crank up the Hz when exclusive/full resolution is lowered (atleast I remember some LCD displays allow it, especially at resolutions in the 640x480 to 800x600 range).

Probably most users on modern hardware probably want non-exclusive full screen, but it would be nice to have this as an option, if you can harangue SDL to do it, @GranMinigun !

**Comment by GranMinigun** [comment]
I'm fairly certain that fullscreen optimizations on Win10+ do not automatically forbid changing display output mode. So running DOSBox on a CRT with original game resolution can be a thing.

It should be possible to "overclock" panels without lowering resolution by overriding vblank timings, making them tighter. That's what I did with my monitor, anyway.

**Comment by johnnovak** [comment]
Not adding support for this as exclusive fullscreen is mostly a thing of the past on Windows, fullscreen and custom refresh rates are a huge rabbit hole on Linux, and there's macOS that surely does something differently too...

VRR monitors are the future, and any custom refresh rate just works on these displays with Staging.

On non-VRR montors, we won't support setting the refresh rate in exclusive fullscreen (and exclusive fullscreen is a bit broken now anyway; it's not recommended to use it).

Similarly, using Staging with CRTs is firmly out of scope too.
